api/portainer.go

```diff
@@ -140,6 +140,7 @@ type (
 		SwarmID     string  `json:"SwarmId"`
 		ProjectPath string
 		Env         []Pair `json:"Env"`
+		Prune       bool   `json:"Prune"`
```

Comment:
IMO, the fact that we need to prune a stack on update should not be stored inside the Stack object. Instead, the `Deploy` method of the `StackManager` interface should accept an extra parameter `prune bool`.